src/utils/object.js

```diff
@@ -44,3 +44,27 @@ export function map(opt_initial) {
 export function hasOwn(obj, key) {
   return hasOwn_.call(obj, key);
 }
+
+/**
+ * Returns the value of the property referenced by dot-separated keys.
+ * e.g.
+ * `getPath('a.b[0].c', {a: {b: [{c: 2}]}}) === 2`
+ *
+ * @param {T} obj a map-like value
+ * @param {string} path a dot-separated list of keys to reference a value
+ * @return {T}
```

Comment:
This should be `*`, it's not guaranteed to be an `Object` (which `T` has to be).